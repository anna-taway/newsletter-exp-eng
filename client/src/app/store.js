import { configureStore } from "@reduxjs/toolkit";
import registrationReducer from "../features/userRegistration/userRegistrationSlice";
import productsReducer from '../features/products/productsSlice';


const store = configureStore({
	reducer: {
		registration: registrationReducer,
		products: productsReducer
	},
});

export default store;