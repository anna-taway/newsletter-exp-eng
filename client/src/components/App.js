import React from 'react';
import { Router } from 'react-router-dom';
import Header from '../components/Header';
import Main from '../components/Main';
import Footer from '../components/Footer';
import history from '../history';



const App = () => {
  return (
    <div>

      <Router history={history}>
        
        <Header />

        <Main />

        <Footer />
       
      </Router>


    </div>
  )
}

export default App;


