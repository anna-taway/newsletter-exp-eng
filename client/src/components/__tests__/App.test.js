import React from "react";
import App from "../App";
import { create } from "react-test-renderer";

import { Provider } from 'react-redux';
import store from '../../app/store';

describe("App component", () => {
  test("App component matches the snapshot", () => {
    const app = create(
      <Provider store={store}>
          <App />
      </Provider>
    );
    expect(app.toJSON()).toMatchSnapshot();
  });
});