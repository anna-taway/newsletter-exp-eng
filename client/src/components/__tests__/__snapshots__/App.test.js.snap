// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App component matches the snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        #Newsletter
      </h6>
    </div>
  </header>
  <div
    className="makeStyles-productsContent-1"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <h1
        className="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Newsletter
      </h1>
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
      >
        Some news about EE and QBE
      </p>
      <div
        className="makeStyles-productsButtons-2"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <a
                  color="inherit"
                  href="/user"
                  onClick={[Function]}
                >
                  Subscribe
                </a>
              </span>
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <a
                  color="inherit"
                  href="/"
                  onClick={[Function]}
                >
                  Find more
                </a>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root makeStyles-cardGrid-3 MuiContainer-maxWidthMd"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    />
  </div>
  <footer
    className="makeStyles-footer-8"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      A production with EE and EBQ
    </h6>
    <p
      className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Contact
    </p>
  </footer>
</div>
`;
